src/pcap/cmd/pcap-bosh-cli/main.go

```diff
@@ -393,31 +417,31 @@ func (e *Environment) connect() error {
 		e.DirectorURL.Path = "/"
 	}
 
+	// If no port was provided, use BOSH default port
+	if e.DirectorURL.Port() == "" {
+		e.DirectorURL.Host = fmt.Sprintf("%s:%d", e.DirectorURL.Host, BoshDefaultPort)
+	}
+
 	if e.DirectorURL.Scheme != "https" {
 		logger.Warn("using unencrypted connection to bosh-director", zap.String("bosh-director-url", e.DirectorURL.String()))
 		e.client = http.DefaultClient
-		return nil
-	}
-
-	logger.Info("using TLS-encrypted connection to bosh-director", zap.String("bosh-director-url", e.DirectorURL.String()))
-	boshCA := x509.NewCertPool()
-	ok := boshCA.AppendCertsFromPEM([]byte(e.CaCert))
-	if !ok {
-		return fmt.Errorf("could not add BOSH Director CA from bosh-config, adding to the cert pool failed")
-	}
-
-	transport := http.DefaultTransport.(*http.Transport).Clone()
-	transport.TLSClientConfig.RootCAs = boshCA
-
-	e.client = &http.Client{
-		Transport: transport,
-	}
+	} else {
+		logger.Info("using TLS-encrypted connection to bosh-director", zap.String("bosh-director-url", e.DirectorURL.String()))
```

Comment:
Why do you want to introduce else block to the condition? The further code is more readable without else block.